friends/src/App.js

```diff
@@ -49,24 +49,24 @@ class App extends Component {
       name: this.state.friendName,
       age: this.state.friendAge,
       email: this.state.friendEmail
-    }
+    };
     axios.post('http://localhost:5000/friends', newfriend)
     .then(result => {
       this.setState({
       friendName: '',
-      friendAge: undefined,
+      friendAge: '',
       friendEmail: '',
-      friends: result.data
-    });
-  })
-  console.log(this.state);
+      friends: result.data,
+      number: newfriend.id
+    })
+  }).then(console.log(this.state));
   }
 
   render() {
     return (
       <div className="App">
+        <Route path='/' render={props => <FriendForm {...props} newFriendName={this.state.friendName} newFriendAge={this.state.friendAge} newFriendEmail={this.state.friendEmail} nameAdd={this.addNameHandler} ageAdd={this.addAgeHandler} emailAdd={this.addEmailHandler} submitFriend={this.addNewFriendHandler} />} />
```

Comment:
Looks as though this one gave you some trouble. A quick tip. Start a new line after each JSX attribute / Props combination. You'll find that it is more readable and easier to spot typos and other varying issues.